### DISCUSSION TITLE
Meaning of CIP connection parameters and connection timeouts?

### DISCUSSION THREAD

**Discussion by kyle-github** [ottowayi/pycomm3#193] [Q&A] [answered]
Hi Ian,

Over on the pylogix site you noted:

>CIPPriority is 0x0A which means the tick time is 1024ms and CIPTimeoutTicks is 0x0E or 14 ticks, so the timeout values is 1024 * 14 = 14336ms or ~14 seconds. The unconnected send method also uses the same values, so it should be the same for unconnected messages as well.

I had seen this broken down a bit differently, with the 0x0A being a tick time and the 0x0E being the number of ticks, before (I do not have the OVDA CIP docs) but I ended up having to use 0xA/0x05 for these two values.   I can't remember what drove that.  Perhaps it was to a PLC-5?

I thought I also read somewhere that this was more or less ignored for Type 3 connections since they were assumed to be non-periodic.  Do you have any insight on that?

As I mentioned over in the other discussion, I found that 5 seconds was the longest I could be safe and that if I let the connection just drop on the PLC side, I got weird and painful behavior (hangs and long delays) on the PC side (mostly Linux).

Due to your comment above, I am going to do more testing (it has been several years) and see what I can come up with.  I would love to change the auto-close time to longer than 5 seconds in my library!

**Comment by ottowayi** [comment, marked as answer]
I'm actually in the process of making and documenting most of the CIP object library and was starting on the connection related objects this afternoon.  I'll try and add some more details on the connection parameters when I get to the connection manager object.   I also want to make the connection parameters more user accessible, but it hasn't been needed yet so I've always found something more important to work on.

I've tested it before and found it actually does matter for explicit messages, connected or unconnected.  It would make sense if you were timing out around 5s and had 0x0A/0x05 parameters. 

Here is how the timeout is calculated:
![image](https://user-images.githubusercontent.com/5274950/143155264-72d13cdd-d3df-474f-802d-ec95da3c6a9c.png)
